bang.action.php

```diff
@@ -56,67 +56,77 @@ public function actReact()
     self::setAjaxMode();
     $cards = explode(';', self::getArg('cards', AT_numberlist, false));
     //		$id = self::getArg( "id", AT_posint, true );
-    $result = $this->game->actReact($cards);
+    $this->game->actReact($cards);
     self::ajaxResponse();
   }
 
   public function actPass()
   {
     self::setAjaxMode();
-    $result = $this->game->actReact(null);
+    $this->game->actReact(null);
     self::ajaxResponse();
   }
 
   public function actCancelPreselection()
   {
     self::setAjaxMode();
-    $result = $this->game->actCancelPreSelection();
+    $this->game->actCancelPreSelection();
     self::ajaxResponse();
   }
 
   public function actSelect()
   {
     self::setAjaxMode();
     $cards = explode(';', self::getArg('cards', AT_numberlist, false));
-    $result = $this->game->select($cards);
+    $this->game->actSelect($cards);
     self::ajaxResponse();
   }
 
   public function actDraw()
   {
     self::setAjaxMode();
     $id = self::getArg('selected', AT_alphanum, true);
-    $result = $this->game->draw($id);
+    $this->game->draw($id);
     self::ajaxResponse();
   }
 
   public function actEndTurn()
   {
     self::setAjaxMode();
-    $result = $this->game->actEndTurn();
+    $this->game->actEndTurn();
     self::ajaxResponse();
   }
 
   public function actCancelEndTurn()
   {
     self::setAjaxMode();
-    $result = $this->game->actCancelEndTurn();
+    $this->game->actCancelEndTurn();
     self::ajaxResponse();
   }
 
   public function actDiscardExcess()
   {
     self::setAjaxMode();
     $cards = array_map('intval', explode(';', self::getArg('cards', AT_numberlist, false)));
-    $result = $this->game->actDiscardExcess($cards);
+    $this->game->actDiscardExcess($cards);
     self::ajaxResponse();
   }
 
   public function actUseAbility()
   {
     self::setAjaxMode();
     $cards = array_map('intval', explode(';', self::getArg('cards', AT_numberlist, false)));
-    $result = $this->game->useAbility($cards);
+    $this->game->useAbility($cards);
+    self::ajaxResponse();
+  }
+
+  public function actChangePref()
+  {
+    self::setAjaxMode();
+    $pref = self::getArg('pref', AT_posint, false);
+    $value = self::getArg('value', AT_posint, false);
+    $player = (int) self::getArg('player', AT_posint, false);
```

Comment:
why do you need that ? can't you use getCurrentPlayerId ?

Reply:
Because you don't need to be a current player to switch this preference. Or is it magically returns a person who did the action?